tests/base.py

```diff
@@ -0,0 +1,208 @@
+import sys
+from dataclasses import dataclass
+from mock import MagicMock, patch
+from typing import Callable
+
+# Prevent importing modules w/Raspi hardware dependencies.
+# These must precede any SeedSigner imports.
+sys.modules['seedsigner.gui.renderer'] = MagicMock()
+sys.modules['seedsigner.gui.screens.screensaver'] = MagicMock()
+sys.modules['seedsigner.views.screensaver'] = MagicMock()
+sys.modules['seedsigner.hardware.buttons'] = MagicMock()
+sys.modules['seedsigner.hardware.camera'] = MagicMock()
+sys.modules['seedsigner.hardware.microsd'] = MagicMock()
+
+from seedsigner.controller import Controller, FlowBasedTestException, StopFlowBasedTest
+from seedsigner.gui.screens.screen import RET_CODE__BACK_BUTTON, RET_CODE__POWER_BUTTON
+from seedsigner.models import Settings
+from seedsigner.views.view import Destination, MainMenuView, View
+
+
+
+class BaseTest:
+
+    @classmethod
+    def setup_class(cls):
+        # Ensure there are no on-disk artifacts after running tests.
+        Settings.SETTINGS_FILENAME = "settings-test.json"
+
+        # Mock out the loading screen so it can't spawn.
+        patch('seedsigner.gui.screens.screen.LoadingScreenThread').start()
+
+
+    @classmethod
+    def teardown_class(cls):
+        pass
+
+
+    @classmethod
+    def reset_settings(cls):
+        """ Wipe and re-initialize the Settings singleton """
+        Settings._instance = None
+        BaseTest.remove_settings()
+
+
+    @classmethod
+    def remove_settings(cls):
+        """ If settings were written to disk, delete """
+        import os
+        try:
+            if os.path.exists(Settings.SETTINGS_FILENAME):
+                os.remove(Settings.SETTINGS_FILENAME)
+        except:
+            print(f"{Settings.SETTINGS_FILENAME} not found to be removed")
```

Comment:
Maybe " could not be removed" would be more accurate since the try block will more likely fail on perms/io errors and not because it wasn't found?